Feature/125 add skos xl

koenedaele opened the pull request.

Wim-De-Clercq commented on tests/test_skos.py:
```diff
@@ -16,77 +14,97 @@
+        l1 = Label('Knokke-Heist', uri='urn:x-skosprovider:gemeenten:Knokke-Heist:nl-BE')
+        l2 = Label('Knokke-Heist', uri='urn:x-skosprovider:gemeenten:Knokke-Heist:nl')
+        l3 = Label('Cnocke-Heyst', type='altLabel', language='vls', uri='urn:x-skosprovider:gemeenten:Knokke-Heist:nl-BE')
+        assert not l1 == l2
```
!=

Wim-De-Clercq commented on tests/test_skos.py:
```diff
@@ -537,46 +522,46 @@ def _get_sortlabel(self):
-        self.assertEqual(None, label([], None))
-        self.assertEqual(None, label([], 'und'))
-        self.assertEqual(None, label([], ['nl-BE']))
+        assert None == label([])
```
All None equalities, or inequalities should probably become `is None` or `is not None`

Wim-De-Clercq reviewed (changes requested):
I know it is just a first iteration of the skos XL.

As I understand it, you could add ANY extra property to labels, not just uri which is a standard one.

If implementors would like to extend this in their own project using `skosprovider`. The `Label` class is easily extended because that's just simple class inheritance. *

The `jsonld` dumper code however, I don't see how they would cleanly extend that to add extra properties without straight up copying a majority of that code. 

This actually is probably true for not just the labels, but any object that could hold extra attributes. The dumper is not very user-friendly extendable.


\* I say that the Label class is easy to extend. But to actually have the provider and rest of the code use your extended Label class is a whole other thing. You then would also need to manage to change the `dict_to_label` to return your extended class. But `dict_to_label` is hardcoded referenced in the `ConceptScheme` class, which is hardcoded referenced in the provider and so forth. So maybe even the label class extensions is not as easy either.
Perhaps extending things is out of scope of the project.

----

Unrelated, but I also noticed that for
```py
def jsonld_conceptscheme_dumper(provider, context = None,
        relations_profile = 'partial', language = 'en'):
```
the `language` parameter is unused. That is at the very least something unusual or suspicious.

koenedaele commented on skosprovider/skos.py:
```diff
@@ -51,18 +54,27 @@ class Label:
 
     def __eq__(self, other):
-        return self.__dict__ == (other if type(other) == dict else other.__dict__)
+        if type(other) == dict:
```
Not really sure if the whole dict thing should still be this prominent in the code. In the early days of skosprovider (probably versie 0.1.0) the provider just returned a dict, not a class. We've always maintained some compatibility there, but I don't think it's really serving a lot of purpose here. But that does seem to be out of scope for this PR and should maybe go into a 2.0.0 version?

koenedaele commented on skosprovider/jsonld.py:
```diff
@@ -299,6 +317,35 @@ def lbl_renderer(l):
+    doc = {
+        'labels_xl': {}
+    }
+    def lbl_xl_renderer(l):
```
I've changed the new short variables that were added in this PR and opened #137 to fix the rest (and figure out a solution for the concept or collection issue.

koenedaele commented on skosprovider/skos.py:
```diff
@@ -51,18 +54,27 @@ class Label:
         if not language:
             language = 'und'
+        if uri and not is_uri(uri):
+            raise ValueError('%s is not a valid URI.' % uri)
```
Changed all the % formatted string, except for the ones in uri.py

koenedaele commented on skosprovider/jsonld.py:
```diff
@@ -299,6 +317,35 @@ def lbl_renderer(l):
     }
     for l in c.labels:
         doc['labels'].setdefault(ltypemap[l.type], []).append(lbl_renderer(l))
+    if not len(doc['labels']):
```
Changed those into early exits.

koenedaele commented on skosprovider/skos.py:
```diff
@@ -76,6 +88,9 @@ def is_valid_type(type):
 
+    def is_xl(self):
+        return self.uri is not None
+
```
Added the URI if present

koenedaele commented:
> Unrelated, but I also noticed that for
> 
> ```python
> def jsonld_conceptscheme_dumper(provider, context = None,
>         relations_profile = 'partial', language = 'en'):
> ```
> 
> the `language` parameter is unused. That is at the very least something unusual or suspicious.

Yes, that's a bug. Fixed it and updated the tests.

koenedaele commented on skosprovider/skos.py:
```diff
@@ -604,7 +622,8 @@ def dict_to_label(dict):
             dict.get('type', 'prefLabel'),
-            dict.get('language', 'und')
+            dict.get('language', 'und'),
+            uri = dict.get('uri')
```
Fixed
